Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java

```diff
@@ -289,6 +319,15 @@ public boolean onTouch(View v, MotionEvent event)
 
     invalidate();
 
+    if (autoDim)
+    {
+      isDim = false;
+      // Start dim timer
+      lastTouch = new Date(System.currentTimeMillis()).getTime();
+      dimHandler.removeCallbacksAndMessages(null);
```

Comment:
This seems to be the most "complete" block of `if (autoDim)`...the one above in the `InputOverlay` ctor does not call `removeCallbacksAndMessages` (assuming because it cannot have any in the ctor) and the one below in `refreshControls` does not set `isDim = false` (not sure if thats a bug).

Is there any reason why this isn't made into a member function and call from all the places?

Reply:
Yeah it should be. I initially only had it after a touch event but realized I had to do the check in other places. Fixed.